lib/connect.c

```diff
@@ -677,7 +677,7 @@ void Curl_updateconninfo(struct connectdata *conn, curl_socket_t sockfd)
 {
   if(conn->transport == TRNSPRT_TCP) {
 #if defined(HAVE_GETPEERNAME) || defined(HAVE_GETSOCKNAME)
-    if(!conn->bits.reuse && !conn->bits.tcp_fastopen) {
+    if(!conn->bits.reuse) {
```

Comment:
This check was done like this already in the first TFO commit 03de4e4b2192. @ghedo, do you recall the  reason? Maybe because the subsequently used functions getpeername and getsockname aren't really reliable on an early TFO connection?

Reply:
I think you should make this change separately. It is not the same change as the header file fix. The cmake fix is very straight-forward, changing this condition is not.

Reply:
done